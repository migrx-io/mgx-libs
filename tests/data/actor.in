rawstring without op
1
{}
{"data": "sting in json data"}
{"context": {"op": "simple", "id": 1}, "data": "simple request"}
{"context": {"op": "sleep", "id": 2}, "data": "sleep request"}
{"context": {"id": 1}, "data": "simple request wihout op"}
{"context": {"id": 2}, "data": "sleep request without op"}
{"context": {"op": "simple", "id": 1}, "data": {"x": 1}}

