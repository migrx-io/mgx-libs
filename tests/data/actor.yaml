desc: | 
  This is example module 

  You need to put here description and main fuctionality description

ops:
  echo:
    desc: |
      Example method do add

      Here we can describe command description

    request:
      meta:
        name:
          required: true
        id:
          required: true
          type: "int"
          choices: [1, 2, 3]
        prefix:
          default: "example"
        email:
          type: "email"
        ids:
          nargs: "*"

      examples: [{"name": "example", "id": 1, "prefix": "test"},
                 {"error": 1},
                 {"error": 1, "name": "exanple", "id": 2},
                 {"name": "exanple", "id": 10},
                 {"name": "example", "id": 1, "email": "asda"},
                 {"name": "example", "id": 1, "email": "main@main.com"},
                 {"name": "example", "id": 1, "ids": "1"},


                ]

    response:
      examples: [{"name": "example", "id": 1, "prefix": "test"}, 
                 {"error": "value name is required"},
                 {"error": "error occured!"},
                 {"error": "value id not in [1, 2, 3]"},
                 {"error": "Invalid email format"},
                 {"name": "example", "id": 1, "email": "main@main.com"},
                 {"error": "value ids not list"},
                ]
