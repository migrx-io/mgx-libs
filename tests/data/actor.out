{"context": {}, "error": "Expecting value: line 1 column 1 (char 0)"}
{"context": {}, "error": "invalid operation: None"}
{"context": {}, "error": "invalid operation: None"}
{"context": {}, "error": "invalid operation: None"}
{"context": {"op": "simple", "id": 1}, "data": "simple:{'db': 'db'}:simple request"}
{"context": {"op": "sleep", "id": 2}, "data": "sleep:{'db': 'db'}:sleep request"}
{"context": {"id": 1}, "error": "invalid operation: None"}
{"context": {"id": 2}, "error": "invalid operation: None"}
{"context": {"op": "simple", "id": 1}, "data": "simple:{'db': 'db'}:{'x': 1}"}

